pkg/controller/controller_instance.go

```diff
@@ -1576,55 +1533,73 @@ func (c *controller) pollServiceInstance(instance *v1beta1.ServiceInstance) erro
 		return c.finishPollingServiceInstance(instance)
 	default:
 		glog.Warning(pcb.Messagef("Got invalid state in LastOperationResponse: %q", response.State))
-		if !time.Now().Before(instance.Status.OperationStartTime.Time.Add(c.reconciliationRetryDuration)) {
-			clone, err := api.Scheme.DeepCopy(instance)
-			if err != nil {
-				return err
-			}
-			toUpdate := clone.(*v1beta1.ServiceInstance)
-			s := "Stopping reconciliation retries on ServiceInstance because too much time has elapsed"
-			glog.Info(pcb.Message(s))
-			c.recorder.Event(instance, corev1.EventTypeWarning, errorReconciliationRetryTimeoutReason, s)
+		if c.reconciliationTimeExpired(instance) {
+			return c.reconciliationTimeExpiredFinishPollingServiceInstance(instance, mitigatingOrphan, provisioning, deleting)
+		}
+		return fmt.Errorf(`Got invalid state in LastOperationResponse: %q`, response.State)
+	}
+	return nil
+}
 
-			if mitigatingOrphan {
-				setServiceInstanceCondition(
-					toUpdate,
-					v1beta1.ServiceInstanceConditionReady,
-					v1beta1.ConditionUnknown,
-					errorOrphanMitigationFailedReason,
-					"Orphan mitigation failed: "+s)
-			} else if deleting || provisioning {
-				setServiceInstanceCondition(toUpdate,
-					v1beta1.ServiceInstanceConditionFailed,
-					v1beta1.ConditionTrue,
-					errorReconciliationRetryTimeoutReason,
-					s)
-			} else {
-				setServiceInstanceCondition(toUpdate,
-					v1beta1.ServiceInstanceConditionReady,
-					v1beta1.ConditionFalse,
-					errorReconciliationRetryTimeoutReason,
-					s)
-			}
+// reconciliationTimeExpired tests if the current Operation State time has
+// elapsed the reconciliationRetryDuration time period
+func (c *controller) reconciliationTimeExpired(instance *v1beta1.ServiceInstance) bool {
```

Comment:
I find I have to think a bit about what this means, but I'm not sure why or how to fix it.

maybe reconcilationTimeoutExpired?

Reply:
Now that you mention it, it does seem a bit funny. How about one of:
`reconciliationTimedOut`
`reconciliationRetryDurationExpired`

Reply:
how about `reconciliationOperationTimeExceeded` ?

Reply:
decided on isReconciliationRetryDurationExceeded, thanks @kibbles-n-bytes